src/irmin/node.ml

```diff
@@ -101,17 +122,27 @@ struct
       t
 
   let remove t k = StepMap.remove k t
-  let default = M.default
-
-  let value_t =
-    let open Type in
-    variant "value" (fun n c x -> function
-      | `Node h -> n h
-      | `Contents (h, m) -> if equal_metadata m M.default then c h else x (h, m))
-    |~ case1 "node" H.t (fun k -> `Node k)
-    |~ case1 "contents" H.t (fun h -> `Contents (h, M.default))
-    |~ case1 "contents-x" (pair H.t M.t) (fun (h, m) -> `Contents (h, m))
-    |> sealv
+  let default = Metadata.default
+
+  module Pre_hash = struct
+    type kind = N | C of Metadata.t
+
+    let kind_t =
+      let open Type in
+      variant "Tree.kind" (fun node contents contents_m -> function
+        | N -> node
+        | C m ->
+            if equal_metadata m Metadata.default then contents else contents_m m)
+      |~ case0 "node" N
+      |~ case0 "contents" (C Metadata.default)
+      |~ case1 "contents" Metadata.t (fun m -> C m)
+      |> sealv
+
+    type entry = { kind : kind; name : Path.step; node : hash }
+    [@@deriving irmin]
+
+    type t = entry list [@@deriving irmin]
+  end
```

Comment:
is this `Pre_hash` module used somewhere? :-)

Reply:
It's not (and in fact has already been removed :-)